src/dynamic_honey_badger/builder.rs

```diff
@@ -112,7 +108,7 @@ where
         let sk_set = SecretKeySet::random(0, rng);
         let pk_set = sk_set.public_keys();
         let sks = sk_set.secret_key_share(0);
-        let sk: SecretKey = rng.gen();
+        let sk = rng.gen::<SecretKey>();
```

Comment:
With the new `distribution`s, should be make it explicit here, which distribution we are using, e.g. something like `Uniform`?

Reply:
I'd say no: `SecretKey` is not a type where any distributions other than a uniform one make any sense.

Reply:
So why not make it easy on auditors by making this explicit?

Reply:
It looks like `Uniform` is only meant for intervals:
[`pub fn new<B1, B2>(low: B1, high: B2) -> Uniform<X>`](https://docs.rs/rand/0.6.1/rand/distributions/struct.Uniform.html#method.new)
And `Fr` (the field of `SecretKey`) doesn't even implement `Ord`.

Reply:
Also, the rand documentation says:
> [Both `Standard` and `Uniform` are in some sense uniform distributions.](https://docs.rs/rand/0.6.1/rand/distributions/index.html#distribution-to-sample-from-a-uniform-range)

Reply:
It's not even inclusive, so we cannot use some sort of `::MAX` here. Alright, I take back the mean things I said!